buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -83,4 +83,25 @@ class PrecommitTasks {
         forbiddenApis.group = "" // clear group, so this does not show up under verification tasks
         return forbiddenApis
     }
+
+    private static Task configureCheckstyle(Project project) {
+        Task checkstyleTask = project.tasks.create('checkstyle')
+        // Apply the checkstyle plugin to create `checkstyleMain` and `checkstyleTest`. It only
+        // creates them if there is main or test code to check and it makes `check` depend
+        // on them. But we want `precommit` to depend on `checkstyle` which depends on them so
+        // we have to swap them.
+        project.pluginManager.apply('checkstyle')
+        project.checkstyle {
+            config = project.resources.text.fromFile(
+                PrecommitTasks.getResource('/checkstyle.xml'), 'UTF-8')
+        }
+        ['checkstyleMain', 'checkstyleTest'].each { taskName ->
```

Comment:
Every time I do this in groovy I look up "groovy foreach loop" and the silly internet gives me back what I did instead of what you did. I like what you did better.